### PR TITLE
Add support for using host network for OpenMPI worker pods

### PR DESCRIPTION
**Pull request by hmizuma**
Some workloads are impacted by overhead of overlay network and I believe can benefit from running on the host network.
Allows passing in `useHostNetwork true` to configure worker pods to run on the host network.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/1805)

### CODE REVIEW

**Comment by jlewi** [comment]
Please sign the CLA

File: kubeflow/openmpi/util.libsonnet [hunk 1]
```diff
@@ -14,4 +14,27 @@
       [std.splitLimit(keyValue, "=", 1)[0]]: std.splitLimit(keyValue, "=", 1)[1]
       for keyValue in $.toArray(str)
     } else {},
+
+  // Convert a string to upper case.
+  upper:: function(x) {
```
Comment (everpeace) [inline kubeflow/openmpi/util.libsonnet]: I think jsonnet has `std.asciiUpper`.  Are there any specific reason to use custom method??

Comment (hmizuma) [reply kubeflow/openmpi/util.libsonnet]: The reason for this was to be consistent with a number of other places where upper and toBool appear such as https://github.com/kubeflow/kubeflow/blob/master/kubeflow/core/util.libsonnet#L15-L22

**Comment by hmizuma** [comment]
I signed the CLA

File: kubeflow/openmpi/util.libsonnet (hunk 1 above)
Comment (jlewi) [reply kubeflow/openmpi/util.libsonnet]: I think std.asciiUpper was added in a later version of jsonnet. We should get rid of our custom function once we a require a version of ksonnet with a sufficiently new version of the jsonnet library.

Please a TODO to that effect.

Comment (jlewi) [reply kubeflow/openmpi/util.libsonnet]: It would be great to file an issue for that and reference the issue in the TODO

**Comment by hmizuma** [comment]
Regarding performance, we were testing Horovod workloads on four p3.16xlarge instances on AWS. Performance over host network was equivalent to running the same work outside of Kubernetes but we saw about 30-60% performance hit when using various CNIs.
We saw good results on kubenet, but it is not available or convenient to configure in all environments. I'll see if I can provide more details and numbers.

File: kubeflow/openmpi/util.libsonnet (hunk 1 above)
Comment (hmizuma) [reply kubeflow/openmpi/util.libsonnet]: I created an issue https://github.com/kubeflow/kubeflow/issues/1826 and updated commit with corresponding TODO

**Comment by everpeace** [comment]
> Performance over host network was equivalent to running the same work outside of Kubernetes but we saw about 30-60% performance hit when using various CNIs.  We saw good results on kubenet, but it is not available or convenient to configure in all environments. 

Wow, It's really interesting.

> I'll see if I can provide more details and numbers.

I'm looking forward to see the numbers.

**Comment by everpeace** [comment]
/assign @pdmack

Would you mind if you took a look this pr and gave an approval?? 🙇 

I reviewed and approved this PR.  But, this contains a tiny change (just add an comment) in `kubeflow/core/util.libsonnet`.  So we need your approval.

Thanks in advance.

**Pull request merged**